apis/python/src/tiledbsoma/sparse_nd_array.py

```diff
@@ -230,23 +236,29 @@ def read_table(self, coords: SparseNdCoordinates) -> Iterator[pa.Table]:
                 schema=A.schema,
             )
 
-            # TODO: make a util function to be shared with DenseNdArray
-            # coords are a tuple of (int or slice-of-int)
-            if len(coords) == 1 and coords[0] == slice(None):
-                # Special case which tiledb-py supports, so we should too
-                coords = coords * A.schema.domain.ndim
-            elif len(coords) != A.schema.domain.ndim:
-                raise ValueError(
-                    f"coordinate length {len(coords)} != array ndim {A.schema.domain.ndim}"
+            # * A sequence of coordinates is accepted, one per dimension.
+            # * Sequence length must be at least one and <= number of dimensions.
+            # * If the sequence contains missing coordinates (length less than number of dimensions),
+            #   then "slice(None)" is assumed for the missing dimensions.
+            # * Per-dimension, explicitly specified coordinates can be one of: a value, a
+            #   list/ndarray/paarray/etc of values, a slice, etc.
+
+            if not (isinstance(coords, list) or isinstance(coords, tuple)):
+                # if not isinstance(coords, tuple):
+                raise SOMAError(
+                    f"coords type {type(coords)} unhandled; expected list or tuple"
+                )
+            if len(coords) < 1 or len(coords) > A.schema.domain.ndim:
+                raise SOMAError(
+                    f"coords {coords} must have length between 1 and ndim ({A.schema.domain.ndim}); got {len(coords)}"
                 )
-            for i in range(A.schema.domain.ndim):
-                coord = coords[i]
-
-                if coord is None:
-                    continue
 
+            for i in range(len(coords)):
                 dim_name = A.schema.domain.dim(i).name
-                if isinstance(coord, int):
+                coord = coords[i]
+                if coord is None:
+                    pass
+                elif isinstance(coord, int):
                     sr.set_dim_points(dim_name, [coord])
                 elif isinstance(coord, list):
```

Comment:
this if/else doesn't handle tuple, which should be a supported coordinate type (just a read-only list)